Wasm-container/README.md

```diff
@@ -0,0 +1,509 @@
+[![License](https://img.shields.io/badge/License-Apache%202.0-orange.svg)](https://opensource.org/licenses/Apache-2.0)
+
+# Vertica Web Assembly UDx experimental container
+
+[Vertica](https://www.vertica.com/) is a massively scalable analytics data warehouse that stores your data and performs analytics on it all in one place.
+
+This repository creates a container image that makes it reasonably easy to experiment with developing User-Defined Extensions (UDxes) for Vertica using WebAssembly.
+
+Vertica has long supported UDxes in a number of languages:
+
+- C++
+- Java
+- Python
+- R
+
+This container creates an environment for creating UDxes in Rust (and,
+as a side-effect, plain-old C).
+
+Java, Python and R UDxes are loaded into auxiliary programs
+implementing the appropriate runtime.  C++ UDxes may be loaded into a
+separate program (known as "fenced"), or may be linked directly into
+the Vertica binary (known as "unfenced", the implication being there's
+nothing between your UDx code and the innards of Vertica).  All Java, Python, and R UDxes are fenced.
+
+Invoking unfenced functions is done by calling a function with
+arguments.  Invoking fenced functions is done through remote procedure
+calls.
+This means 
+unfenced UDxes have a performance advantage over fenced UDxes.  There is a corresponding savings in the cost of data movement (data are pushed onto or popped off of the stack instead of transmitted through a socket).  However, unfenced UDxes are risky: a bug in an unfenced UDx risks crashing the Vertica server.
+
+Safe programming in C++ can be a challenge.
+
+Enter Web Assembly (Wasm).  Wasm code runs in a sandbox with
+clearly-defined interaction between the Wasm and the host program.
+Wasm code accesses memory in its sandbox, and cannot access memory
+outside its sandbox.  The host program controls what enters and leaves
+the sandbox.
+
+The Web Assembly sandbox provides most of the safety guarantees we get
+from executing a UDx in a separate process using remote procedure
+calls. Supporting Web Assembly UDxes lets us support unfenced UDxes in
+any language that can compile to Wasm, confident that the user-defined
+code cannot corrupt Vertica's data structures.
+
+For details about developing Vertica UDxes, see [Extending Vertica](https://www.vertica.com/docs/latest/HTML/Content/Authoring/ExtendingVertica/ExtendingVertica.htm).
```

Comment:
Probably better to link directly to https://www.vertica.com/docs/12.0.x/HTML/Content/Authoring/ExtendingVertica/UDx/DevelopingUDxs.htm .